tests/components/ChatPanel.test.tsx

```diff
@@ -188,3 +199,101 @@ it("applies link target setting (default blank)", async () => {
   render(<ChatPanel />);
   expect(screen.getByText("msg link")).toHaveAttribute("target", "_blank");
 });
+
+describe("loadSessionState works as expected", () => {
+  const jestHostname = "localhost";
+  window.location.hostname = jestHostname;
+  const mockConvoId = "dummy-id";
+  const mockKey = getStateLocalStorageKey(jestHostname, mockConvoId);
+  const mockPanelState: PanelState = {
+    scrollPosition: 23,
+  };
+  const mockConvoState = {
+    conversation: {
+      conversationId: mockConvoId,
+      messages: [{ ...dummyMessage, timestamp: new Date().toISOString() }],
+    },
+  };
+
+  it("saves panel state to local storage", () => {
+    mockChatState(mockConvoState);
+    const storageSetSpy = jest.spyOn(Storage.prototype, "setItem");
+
+    const { container } = render(<ChatPanel />);
+    const scrollDiv = getChatPanelScrollDiv(container);
+
+    fireEvent.scroll(scrollDiv, {
+      target: { scrollTop: mockPanelState.scrollPosition },
+    });
+
+    expect(storageSetSpy).toHaveBeenCalledWith(
+      mockKey,
+      JSON.stringify(mockPanelState)
+    );
+    expect(localStorage.getItem(mockKey)).toEqual(
+      JSON.stringify(mockPanelState)
+    );
+  });
+
+  it("loads panel from local storage", () => {
+    mockChatState(mockConvoState);
+    localStorage.setItem(mockKey, JSON.stringify(mockPanelState));
+    const storageGetSpy = jest.spyOn(Storage.prototype, "getItem");
+
+    render(<ChatPanel />);
+    expect(storageGetSpy).toHaveBeenCalledWith(mockKey);
+    expect(localStorage.getItem(mockKey)).toEqual(
+      JSON.stringify(mockPanelState)
```

Comment:
we should verify the panel is scrolled to the saved location

Reply:
did you try this? https://stackoverflow.com/questions/43148351/how-to-spy-on-window-scrollto-in-jest